File: core/autonomy/semantic_goal_engine.py
from __future__ import annotations

from typing import Any, Dict


MAX_GOAL_SIZE = 4096


def resolve_semantic_goal(
    payload: Dict[str, Any],
) -> Dict[str, Any]:

    goal = str(
        payload.get("goal", "")
    )[:MAX_GOAL_SIZE]

    return {
        "goal": goal,
        "resolved": bool(goal),
        "priority": int(
            payload.get("priority", 1)
        ),
    }

File: src/autonomy/semanticGoalEngine.ts
/**
 * Converted from Python: core/autonomy/semantic_goal_engine.py
 * @generated — WebWeaveX python→javascript library port
 */

import * as py from "../runtime/pyCompat.js";

export let MAX_GOAL_SIZE: any = 4096;
export function resolveSemanticGoal(payload: any): any {
  var goal: any = py.slice(py.toStr(py.get(payload, "goal", "")), null, MAX_GOAL_SIZE);
  return {"goal": goal, "resolved": py.truthy(goal), "priority": py.toInt(py.get(payload, "priority", 1))};
}
